server/core/config/raw/deployment.go

```diff
@@ -0,0 +1,27 @@
+package raw
+
+import (
+	validation "github.com/go-ozzo/ozzo-validation"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+)
+
+type Deployments struct {
+	StorageBackend *StorageBackend `yaml:"storage-backend" json:"storage-backend"`
+}
+
+func (d Deployments) Validate() error {
+	return validation.ValidateStruct(&d,
+		validation.Field(&d.StorageBackend),
+	)
+}
+
+func (d *Deployments) ToValid() valid.Deployments {
+	if d.StorageBackend == nil {
+		return valid.Deployments{}
+	}
+
+	storageBackend := d.StorageBackend.ToValid()
+	return valid.Deployments{
+		StorageBackend: &storageBackend,
+	}
+}
```

Comment:
This got me thinking that jobs and deployments should probably be coalesced since it's a lot easier to just configure one bucket for the whole service and separate different functions by prefix.

ie.

```
deploy-artifacts-bucket
   /jobs/
   /deploys/
```

maybe configuration would just be something generic like what they do for temporal here:
https://docs.temporal.io/references/configuration/#persistence

open to ideas.

Reply:
Yeah I agree! What do you think of this setup: 
```
persistence:
    deploy: default
    jobs: default
    prefix: deploy-artifacts-bucket
    datastores:
        default:
            s3:
                bucket-name: <name-of-bucket>
```